src/libraries/System.Data.Common/src/System/Data/Common/DbConnectionStringBuilder.cs

```diff
@@ -393,6 +391,9 @@ internal Attribute[] GetAttributesFromCollection(AttributeCollection collection)
             return attributes;
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2112",
```

Comment:
It's a bit odd that we have this suppression on the whole class, but we also need it on each method.

Reply:
The class suppression is for IL2113 which specifically is about members of base classes or interfaces, in this case ICustomTypeDescriptor. For members of this class we put the warning on the individual members so that they can be suppressed with more granularity, but it would also be possible to suppress IL2112 on the whole class if we prefer that approach.

Reply:
Ah, I missed they were different codes. Maybe the "short name" would help here to make it more obvious.

>  it would also be possible to suppress IL2112 on the whole class if we prefer that approach.

No, I think what you have is fine.